R/preprocessing.R

```diff
@@ -2477,12 +2530,26 @@ ReadVizgen <- function(
   if (use.dir && !dir.exists(paths = data.dir)) {
     stop("Cannot find Vizgen directory ", data.dir)
   }
-  # Identify input files
-  files <- c(
-    transcripts = transcripts %||% 'cell_by_gene[_a-zA-Z0-9]*.csv',
-    spatial = spatial %||% 'cell_metadata[_a-zA-Z0-9]*.csv',
-    molecules = molecules %||% 'detected_transcripts[_a-zA-Z0-9]*.csv'
-  )
+  
+  # use Cellpose output ----
+  if (use.cellpose.out && grep("Cellpose", list.files(data.dir)) %>% any) {
+    message("Cellpose output will be used..")
+    # Identify input files
+    files <- c(transcripts = transcripts %||% paste0(data.dir, 
+                                                     paste0("/", list.files(data.dir, pattern = "Cellpose"), 
+                                                            "/cellpose_cell_by_gene.csv")),
+               
+               spatial = spatial %||% paste0(data.dir, 
+                                             paste0("/", list.files(data.dir, pattern = "Cellpose"), 
+                                                    "/cellpose_cell_metadata.csv")),
+               
+               molecules = molecules %||% "detected_transcripts[_a-zA-Z0-9]*.csv")
```

Comment:
I have only seen the Vizgen/cellpose output files for a single experiment, but in that case cellpose_ was not a prefix for the cell_by_gene.csv and cell_metadata.csv files. Do you feel we need to include the prefix, or can we relax the file naming to match `*cell_by_gene.csv` and `*cell_metadata.csv`?

Reply:
We working with Vizgen, Resolve, Xenium technologies. The last data we had from Vizgen looked like this:
```
region_0/
├── xxx_CellBound-VA292-BW_VMSC04701_region_0.vzg
├── Cellpose_polyT_CB3
├── detected_transcripts.csv
├── images
└── summary.png

region_0/Cellpose_polyT_CB3/
├── xxx_rCellBound-VA292-BW_VMSC04701_region_0_cellpose.vzg
├── cellpose_cell_by_gene.csv
├── cellpose_cell_metadata.csv
├── cellpose_micron_space.parquet
└── cellpose_mosaic_space.parquet
```
they seem to keep the `cellpose_` prefix. But I will check again with the latest Merscope instrument output.

Reply:
Good to know. I think we should be safe updating the file matching to `*cell_by_gene.csv` and `*cell_metadata.csv`, and I'll make note of this assumption in the docs so users are aware.

Reply:
The addition of "cellpose_" at the beginning of the file name was an older version of the data from Vizgen. It's something that should be accounted for but it is not the current format of the data.